src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -175,12 +178,15 @@ private void linkToCaseWithoutRevoking(
     }
 
     private List<Claim> getAllCasesBy(User user, ImmutableMap<String, String> searchString) {
-        return extractClaims(searchAll(user, searchString));
+        return extractClaims(asStream(searchAll(user, searchString))
+            .filter(isAwaitingCitizenState.negate())
```

Comment:
I am thinking, wouldn't this be better if we pass this as filter as search criteria to CCD. Probably need to give it a try whether it works. 
Please refer here:
https://github.com/hmcts/cmc-claim-store/blob/2ab561410fa921b96376e01c05d85dca285f5823/src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java#L337

Reply:
👍 I'll take a look and see if it works. Thanks!